processing/data_collection/gazette/spiders/pe_recife.py

```diff
@@ -51,7 +51,7 @@ def start_requests(self):
         for date in dates:
             yield scrapy.Request(
                 url=self.EDITIONS_IN_DATE_URL.format(full_date=date.strftime("%Y%m%d")),
```

Comment:
This spider is one the best spiders to filter by date as we have a one-to-one relation between a gazette and a request. So an easiest way to filter is just to create the `dates` list using `start_date` as `dtstart`. No need for Middlewares and the code will be much more clearer.

Reply:
I kind of agree here. But consider my point about adding some other argument like `end_date` or something else we haven't thought about and having to refactor every spider again.